abi_wrapper.go

```diff
@@ -0,0 +1,605 @@
+package web3
+
+import (
+	"errors"
+	"io"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/gochain-io/gochain/v3/accounts/abi"
+)
+
+func ABIBuiltIn(contractFile string) (io.Reader, error) {
+	if val, ok := bundledContracts[contractFile]; ok {
+		return strings.NewReader(val), nil
+	}
+	return nil, errors.New("Cannot find bundled contract")
+}
+
+func ABIOpenFile(contractFile string) (io.Reader, error) {
+	if _, err := os.Stat(contractFile); os.IsNotExist(err) {
+		return nil, err
+	}
+	jsonReader, err := os.Open(contractFile)
+	if err != nil {
+		return nil, err
+	}
+	return jsonReader, nil
+}
+
+func GetAbi(contractFile string) *abi.ABI {
```

Comment:
I was thinking that the `ABI*` functions would return `(*abi.ABI, error)` and this `GetAbi` function would go away completely (or become a very small helper within `cmd/web`). Either way, we should be returning errors, never logging, and especially never exiting the process from within this package.